src/main/java/software/amazon/cloudwatchlogs/emf/logger/MetricsLogger.java

```diff
@@ -36,6 +38,8 @@ public class MetricsLogger {
     private CompletableFuture<Environment> environmentFuture;
     private EnvironmentProvider environmentProvider;
 
+    @Getter @Setter private boolean flushPreserveDimensions = true;
```

Comment:
I find the naming here a bit confusing/long. Since the default is to keep dimensions why not name it `preserveDimensions = false;`

Reply:
Discussed privately. Would keep the naming & initialization value should be "true".